modules/remote/remote_application_utils.lua

```diff
@@ -0,0 +1,122 @@
+--- Module which provides utils interface for application management on remote host
+--
+-- *Dependencies:* none
+--
+-- *Globals:* none
+-- @module RemoteAppUtils
+-- @copyright [Ford Motor Company](https://smartdevicelink.com/partners/ford/) and [SmartDeviceLink Consortium](https://smartdevicelink.com/consortium/)
+-- @license <https://github.com/smartdevicelink/sdl_core/blob/master/LICENSE>
+
+local constants = require("remote/remote_constants")
+
+local RemoteAppUtils = {
+	mt = { __index = {} }
+}
+
+local function HandleResult(isResultBoolean, result, data)
+  local isResultValid = false
+  local isSuccess
+  if result == constants.ERROR_CODE.SUCCESS then
+    isResultValid = true
+    isSuccess = true
+  else
+    isSuccess = false
+    for _, v in pairs(constants.ERROR_CODE) do
+      if result == v then
+        isResultValid = true
+        break;
+      end
+    end
+  end
+
+  if not isResultValid then
+    error("Module RemoteAppUtils received invalid result value: " .. result)
+  end
+  if isResultBoolean == true then
+    return isSuccess, isSuccess
+  end
+  return isSuccess, data
+end
+
+--- Module which provides utils interface for application management on remote host
+-- @type RemoteAppUtils
+
+--- Construct instance of RemoteAppUtils type
+-- @tparam RemoteConnection connection RemoteConnection instance
+-- @treturn RemoteAppUtils Constructed instance
+function RemoteAppUtils.RemoteAppUtils(connection)
+  local res = { }
+  res.connection = connection:GetConnection()
+  setmetatable(res, RemoteAppUtils.mt)
+  return res
+end
+
+--- Check connection
+-- @treturn boolean Return true in case connection is active
+function RemoteAppUtils.mt.__index:Connected()
+  return self.connection:connected()
+end
+
+--- Start application on remote host
+-- @tparam string remotePathToApp Path to application file on remote host
+-- @tparam string appName Name of application executive file
+-- @treturn boolean Return true in case of success
+function RemoteAppUtils.mt.__index:StartApp(remotePathToApp, appName)
```

Comment:
@jacobkeeler No, it can be used to control other applications on SDL host (for example HMI or proprietary headunit system)
But within current proposal it is used for controlling SDL Core only.